napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, Iterable
+
+from qtpy.QtCore import QModelIndex, Qt
+
+from napari._qt.containers.qt_list_model import QtListModel
+from napari.components import Dims
+from napari.utils.events import SelectableEventedList
+
+
+class AxisModel:
+    """View of an axis within a dims model.
+
+    The model keeps track of axis names and allows read / write
+    acces on the cossesponding rollable state of a Dims object.
+
+    Parameters
+    ----------
+    dims : napari.components.dims.Dims
+        Parent Dims object.
+    axis : int
+        Axis index.
+
+    Attributes
+    ----------
+    dims : napari.components.dims.Dims
+        Dimensions object modeling slicing and displaying.
+    axis : int
+        Axis index.
+    """
+
+    def __init__(self, dims: Dims, axis: int) -> None:
+        self.dims = dims
+        self.axis = axis
+
+    def __hash__(self) -> int:
+        return id(self)
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        return self.dims.axis_labels[self.axis]
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, int):
+            return self.axis == other
+        if isinstance(other, str):
+            return repr(self) == other
+        if isinstance(other, AxisModel):
+            return (self.dims is other.dims) and (self.axis == other.axis)
+        return NotImplemented
+
+    @property
+    def rollable(self) -> bool:
+        """
+        If the axis should be rollable.
+        """
+        return self.dims.rollable[self.axis]
+
+    @rollable.setter
+    def rollable(self, value: bool) -> None:
+        rollable = list(self.dims.rollable)
+        rollable[self.axis] = value
+        self.dims.rollable = tuple(rollable)
+
+
+class AxisList(SelectableEventedList[AxisModel]):
+    def __init__(self, axes: Iterable[AxisModel]):
+        super().__init__(axes)
+
+    @classmethod
+    def from_dims(cls, dims: Dims) -> 'AxisList':
```

Comment:
Maybe use `Self` from `typing_extension` as it will be more correct.

Reply:
I was not aware of `Self`. Agreed!

Reply:
Done. Commit coming soon.